Training2/thongnq-training2/src/App.js

```diff
@@ -0,0 +1,53 @@
+import React, {useState, useRef, useEffect} from "react";
+
+import Button from "./task3/Button";
+
+function App() {
+
+  const [userInput, setUserInput] = useState(0);
+
+  const [error, setError] = useState("");
+
+  const [stopTimer, setStopTimer] = useState(false);
+
+  useEffect(() => {
+    const timer = userInput > 0 && !stopTimer && setInterval(() => setUserInput(userInput - 1), 1000);
+    return () => clearInterval(timer);
+  }, [userInput]);
+
+  const inputRef = useRef(null);
+
+  const handleStartCounter = () => {
+    const userInput = inputRef.current.value;
+    if(userInput.trim === ""){
+      setError("Please input a number");
+    } else if(isNaN(userInput)){
+      setError("Invalid input. Must be a number");
+    } else if (parseInt(Number(userInput)) != userInput){
+      setError("Invalid input. Must be a integer");
+    } else if (parseInt(userInput) <= 0){
+      setError("Number must be greater than 0");
+    } else {
+      setUserInput(parseInt(userInput));
```

Comment:
@Thong-Nguyen-Zigvy 
when you separate set userInput & setStopTimer, the useEffect will run twice, and we don't sure useEffect run with stopTimer and userInput with the newest value, you and group them into one state